src/tests/java/htsjdk/samtools/SamReaderFactoryTest.java

```diff
@@ -132,8 +132,14 @@ public void samRecordFactoryTest(final String inputFile) throws IOException {
     public Object[][] composeAllPermutationsOfSamInputResource() {
         final List<SamInputResource> sources = new ArrayList<SamInputResource>();
         for (final InputResource.Type dataType : InputResource.Type.values()) {
+            if (dataType.equals(InputResource.Type.SRA_ACCESSION))
+                continue;
+
             sources.add(new SamInputResource(composeInputResourceForType(dataType, false)));
             for (final InputResource.Type indexType : InputResource.Type.values()) {
+                if (indexType.equals(InputResource.Type.SRA_ACCESSION))
+                    continue;
+
```

Comment:
I can, but I won't be able to preserve the same header (at least reference names will be canonical names, not short names). Also, record order might be different because SRA stores records in a different way

It is a little different from other tests cases here, since SRA is not only a different protocol of getting data, but also a different format of representing that data